Netable/Netable/GraphQLRequest.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  GraphQLRequest.swift
+//  Netable
+//
+//  Created by Brendan on 2022-09-12.
+//  Copyright © 2022 Steamclock Software. All rights reserved.
+//
+
+import Foundation
+
+public protocol GraphQLRequest: Request {
+    associatedtype Parameters = [String: String]
+
+    func getGraphQLQueryFileContents() -> String
+}
+
+public extension GraphQLRequest {
+    var method: HTTPMethod { HTTPMethod.post }
+
+    var path: String { "" }
+
+    func getGraphQLQueryFileContents() -> String {
+        guard let resource = "\(type(of: self))".split(separator: ".").last,
```

Comment:
This makes me a little uneasy, I'm pretty sure that the function that is ultimately producing the name of the type here is `String.init(describing:)` whose description of the possible return values (https://developer.apple.com/documentation/swift/string/init(describing:)-67ncf) contains this troubling phase: "An unspecified result is supplied automatically by the Swift standard library.", meaning I don't think this is guaranteed to produce something in the format we currently are getting in all cases, and it could theoretically change at some point.

I could probably live with this hack in an application, because the reality is even though it isn't documented they are very unlikely to change it, and if there are edge cases that cause this to be an unexpected value, they would be less likley to pop up unexpectedly inside an application vs a library included in someone else's build process. I think for a library, it's a little risky, though. 

At the very least, if we were going to support this as an option, I'd want to have some OTHER ways to set that as well, probably as a string literal, a file in the bundle, and a file at an arbitrary path. And preferably make people opt in to the slightly risky type name dissembly behaviour.

Maybe something like 

```
enum GraphQLQuerySource {
    case literal(String) // full text string of the query
    case resource(String) // name of a .graphql file in the app bundle
    case file(URL) // arbitrary file URL
    case autoResource // autogenerate resource file from dynamic class name (risky)
}

public protocol GraphQLRequest: Request {
     associatedtype Parameters = [String: String]
     var source: GraphQLQuerySource
 }
```